src/std/crypto/pkey.ss

```diff
@@ -21,10 +26,116 @@
     (EVP_PKEY_keygen ctx)
     (foreign-release! ctx)))
 
-(def (bytes->private-key type bytes engine: (engine #f))
+;;; * Meta: pkey-[type|type-id|is-a?|?]
+
+(def pkey-type EVP_PKEY_get0_type_name)
+(def pkey-type-id EVP_PKEY_get_base_id)
+(def pkey-is-a? EVP_PKEY_is_a)
+(def (pkey? k) (and (EVP_PKEY? k) #t))
+
+;;; * EVP_PKEY_* Aliases: 'FOO = EVP_PKEY_FOO
+
+(def evp-pkey-alias-table (make-hash-table))
+
+(def (evp-pkey-alias name)
+  (def (find-alias)
+    (ignore-errors
+     (eval (string->symbol (format "std/crypto/libcrypto#EVP_PKEY_~a" name)))))
```

Comment:
please don't eval here, it is unnecessary and it also at odds with the tree shaker; there are much better ways to do that -- preferably with a macro that just does the right thing and defines things in advance..